Essentials/src/main/java/com/earth2me/essentials/commands/Commandtpaccept.java

```diff
@@ -18,60 +24,105 @@ public Commandtpaccept() {
 
     @Override
     public void run(final Server server, final User user, final String commandLabel, final String[] args) throws Exception {
-        final User requester;
-        try {
-            requester = ess.getUser(user.getTeleportRequest());
-        } catch (final Exception ex) {
+        final boolean excludeHere;
+        if (args.length > 0) {
+            excludeHere = args[0].equals("*") || args[0].equalsIgnoreCase("all");
+        } else {
+            excludeHere = false;
+        }
+
+        if (!user.hasPendingTpaRequests(true, excludeHere)) {
             throw new Exception(tl("noPendingRequest"));
         }
 
-        if (!requester.getBase().isOnline()) {
+        if (args.length > 0) {
+            if (excludeHere) {
+                IUser.TpaRequest request;
+                int count = 0;
+                while ((request = user.getNextTpaRequest(true, true, true)) != null) {
+                    try {
+                        handleTeleport(user, request, commandLabel);
+                        count++;
+                    } catch (Exception e) {
+                        ess.showError(user.getSource(), e, commandLabel);
+                    } finally {
+                        user.removeTpaRequest(request.getName());
+                    }
+                }
+                user.sendMessage(tl("requestAcceptedAll", count));
+                throw new NoChargeException();
```

Comment:
There's a lot going on in this method - pull this out into a `acceptAllRequests` method perhaps?